File: app/src/main/java/com/rolandvitezhu/todocloud/dialog/SortTodoListDialog.java
package com.rolandvitezhu.todocloud.dialog;

import android.app.Dialog;
import android.content.Context;
import android.support.annotation.NonNull;
import android.view.Window;

import com.rolandvitezhu.todocloud.R;

import butterknife.ButterKnife;
import butterknife.OnClick;

public class SortTodoListDialog extends Dialog {

  public interface Presenter {
    void onSortByDueDatePushed();

    void onSortByPriorityPushed();
  }

  private Presenter presenter;

  public SortTodoListDialog(@NonNull Context context,
                           Presenter presenter) {
    super(context);
//    BaseInjector.getComponent().inject(this);
    this.presenter = presenter;

    requestWindowFeature(Window.FEATURE_NO_TITLE);
    setContentView(R.layout.dialog_sorttodolist);
    ButterKnife.bind(this);
    setCancelable(true);

    show();
  }

  @OnClick(R.id.dialog_sort_by_due_date)
  public void onSortByDueDateClicked() {
    presenter.onSortByDueDatePushed();
    dismiss();
  }

  @OnClick(R.id.dialog_sort_by_priority)
  public void onSortByPriorityClicked() {
    dismiss();
    presenter.onSortByPriorityPushed();
  }

}

File: app/src/main/java/com/rolandvitezhu/todocloud/ui/activity/main/dialog/SortTodoListDialog.kt
package com.rolandvitezhu.todocloud.ui.activity.main.dialog

import android.app.Dialog
import android.content.Context
import android.view.Window
import androidx.databinding.DataBindingUtil
import com.rolandvitezhu.todocloud.R
import com.rolandvitezhu.todocloud.databinding.DialogSorttodolistBinding

class SortTodoListDialog(context: Context,
                         private val presenter: Presenter) : Dialog(context) {
    interface Presenter {
        fun onSortByDueDatePushed()
        fun onSortByPriorityPushed()
    }

    fun onSortByDueDateClicked() {
        presenter.onSortByDueDatePushed()
        dismiss()
    }

    fun onSortByPriorityClicked() {
        dismiss()
        presenter.onSortByPriorityPushed()
    }

    init {
        //    BaseInjector.getComponent().inject(this);
        requestWindowFeature(Window.FEATURE_NO_TITLE)

        val dialogSorttodolistBinding: DialogSorttodolistBinding =
                DataBindingUtil.inflate(
                        layoutInflater,
                        R.layout.dialog_sorttodolist,
                        null,
                        false)
        setContentView(dialogSorttodolistBinding.root)
        dialogSorttodolistBinding.sortTodoListDialog = this

        setCancelable(true)
        show()
    }
}